python/feathub/processors/flink/flink_types_utils.py

```diff
@@ -92,6 +98,75 @@ def to_flink_type(feathub_type: DType) -> DataType:
     )
 
 
+def cast_field_type(
+    table: NativeFlinkTable, field_name_and_types: Dict[str, DataType]
+) -> NativeFlinkTable:
+    """
+    Converts given fields in the flink table to the dedicated data types.
+
+    Note that this method would not change the nullable information of the fields.
+
+    :param table: the Flink table containing the fields to be casted
+    :param field_name_and_types: a dict containing the names of the fields and the
+                                 target data types to cast.
+    """
+    expressions = []
+    for field_name, field_type in field_name_and_types.items():
+        current_type = table.get_schema().get_field_data_type(field_name)
+        result_type = _adjust_flink_type_nullable_information(field_type, current_type)
+        if result_type == current_type:
+            continue
+        expressions.append(
+            native_flink_expr.col(field_name).cast(result_type).alias(field_name)
```

Comment:
We need to rename the field with alias(...) otherwise a new field would be appended to the table. Take `test_transform_with_expr_feature_after_sliding_feature` as an example, after casting the field `total_cost` to Double without alias, the result table would be as follows.
```
(
  `name` STRING NOT NULL,
  `first_time` STRING,
  `last_time` STRING,
  `cnt` BIGINT,
  `__event_time_attribute__` TIMESTAMP_LTZ(3) *ROWTIME*,
  `window_time` BIGINT,
  `total_time` BIGINT,
  `total_time-DOUBLE` DOUBLE
)
```

Reply:
Sounds good.